libs/community/langchain_community/tools/azure_ai_services/azure_file_translation.py

```diff
@@ -0,0 +1,162 @@
+                endpoint=azure_translate_endpoint,
+                credential=AzureKeyCredential(azure_translate_key),
+            )
+
+        except ImportError:
+            raise ImportError(
+                "azure-ai-translation-text is not installed. "
+                "Run `pip install azure-ai-translation-text` to install."
+            )
+
+        return values
+
+    def _read_text_from_file(self, file_path: str) -> str:
+        """
+        Read and return text from the specified file,
+        supporting PDF, DOCX, PPTX, XLSX, HTML, and XML formats.
+
+        Args:
+            file_path (str): Path to the input file.
+
+        Returns:
+            str: Extracted text from the file.
+
+        Raises:
+            ValueError: If the file type is unsupported.
+        """
+
+        file_extension = os.path.splitext(file_path)[1].lower()
+
+        # Map file extensions to loader classes
+        loader_map = {
+            ".pdf": UnstructuredPDFLoader,
+            ".docx": UnstructuredWordDocumentLoader,
+            ".pptx": UnstructuredPowerPointLoader,
+            ".xlsx": UnstructuredExcelLoader,
+            ".xml": UnstructuredXMLLoader,
+            ".html": UnstructuredHTMLLoader,
+        }
+
+        loader_class = loader_map.get(file_extension)
+
+        if file_extension == ".txt":
+            # Handle plain text files directly
+            return self._read_text(file_path)
+        elif loader_class is None:
+            raise ValueError(f"Unsupported file type: {file_extension}")
+
+        # Load the document using the appropriate loader
+        loader = loader_class(file_path)
+        data = loader.load()
+
+        return " ".join([doc.page_content for doc in data])
+
+    def _read_text(self, file_path: str) -> str:
+        """Read text from a plain text file."""
+        with open(file_path, "r", encoding="utf-8") as file:
+            return file.read().strip()
+
+    def _translate_text(self, text: str, target_language: Optional[str] = None) -> str:
+        """
+        Translate the input text to the target language
+        using the Azure Text Translation API.
+
+        Args:
+            text (str): The text to be translated.
+            target_language (str, optional):
+                The target language for translation (default: Spanish).
+
+        Returns:
+            str: Translated text.
+
+        Raises:
+            RuntimeError: If the translation request fails.
+        """
+        if target_language is None:
+            target_language = self.target_language
+
+        try:
+            from azure.ai.translation.text.models import InputTextItem
+        except ImportError:
+            raise ImportError("Run 'pip install azure-ai-translation-text'.")
+
+        try:
+            request_body = [InputTextItem(text=text)]
+            response = self.translate_client.translate(
+                content=request_body, to=[target_language]
```

Comment:
Which version of the text translation library are you using? I believe these parameters passed should be called the following:

```suggestion
                body=request_body, to_language=[target_language]
```